metricbeat/internal/metrics/opt.go

```diff
@@ -0,0 +1,78 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package metrics
+
+// OptUint is a wrapper for "optional" types, with the bool value indicating
+// if the stored int is a legitimate value.
+type OptUint struct {
+	exists bool
+	value  uint64
+}
+
+// NewNone returns a new OptUint wrapper
+func NewNone() OptUint {
+	return OptUint{
+		exists: false,
+		value:  0,
+	}
+}
+
+// NewValue returns a new OptUint wrapper with a given int
+func NewValue(i uint64) OptUint {
```

Comment:
nit; Names communicate intention.  To me a name like `New...` communicates that a pointer is returned.

Reply:
Do you have any other ideas? Maybe just `Value(i uint64)` and `None()` or something?

Reply:
@fearful-symmetry what do you think about this proposal by Steffen? https://github.com/elastic/beats/pull/25771#discussion_r646855418

With this we would have `OptUintWith(i uint64)` and the zero value could be obtained with `var OptUint.../OptUint{}`. And setters wouldn't be needed, you could directly assign.